Source/Core/DolphinQt/Config/Mapping/IOWindow.cpp

```diff
@@ -173,6 +211,17 @@ void IOWindow::OnDialogButtonPressed(QAbstractButton* button)
   }
 
   m_reference->SetExpression(m_expression_text->toPlainText().toStdString());
+  m_controller->UpdateReferences(g_controller_interface);
+  m_apply_button->setText(m_apply_button->text().remove(QStringLiteral("*")));
+
+  if (ciface::ExpressionParser::ParseStatus::SyntaxError == m_reference->GetParseStatus())
+  {
+    QMessageBox error(this);
```

Comment:
This should probably be modal, right? See also #7735

Reply:
:+1: